libsolidity/codegen/ExpressionCompiler.cpp

```diff
@@ -1487,6 +1491,13 @@ void ExpressionCompiler::appendArithmeticOperatorCode(Token::Value _operator, Ty
 		break;
 	case Token::Mul:
 		m_context << Instruction::MUL;
```

Comment:
I'm not 100% sure this works correctly for signed 128 bit numbers. Could you add a test for that, please? We especially need to consider cases where the number has some dirty higher order bits. Perhaps we should clean the numbers just in case.

Reply:
Added the cleanup.